src/utility/bundleInjector.js

```diff
@@ -0,0 +1,78 @@
+import { filter, map } from 'lodash';
+
+const isGlobalAvailable = global => !!window[global];
+
+const startAvailabilityInterval = (global, resolve) => {
+  const availabilityInterval = setInterval(() => {
+    if (isGlobalAvailable(global)) {
+      resolve(global);
+      clearInterval(availabilityInterval);
+    }
+  }, 100);
+
+  return availabilityInterval;
+};
+
+const addBundleItem = (item, type, bundleName, bundleBaseUrl) => {
+  const attr = type === 'script' ? 'src' : 'href';
+  const itemName = type === 'script' ? item.script : item;
+  const itemGlobal = item.global;
+  let checkInterval = null;
+
+  const script = document.createElement(type);
+  script.setAttribute(attr, `${bundleBaseUrl}/bundles/${bundleName}/latest/${itemName}`);
+
+  if (type === 'script') {
+    document.body.appendChild(script);
+  } else {
+    script.rel = 'stylesheet';
+    script.type = 'text/css';
+    document.head.appendChild(script);
+  }
+
+  return new Promise((resolve, reject) => {
+    script.onload = () => {
+      if (type === 'link' || isGlobalAvailable(itemGlobal)) {
+        resolve(itemGlobal);
+      } else {
+        checkInterval = startAvailabilityInterval(itemGlobal, resolve);
+      }
+    };
+
+    script.onerror = () => {
+      if (checkInterval) clearInterval(checkInterval);
+      reject(bundleName);
+    };
+  });
+};
+
+export default {
+  /**
+    * Load bundles method.
+    * @param {object} config - Config object width properties:
+    * scriptType, baseURL, styles, scripts.
+    * @param {object} bundle - Bundle object that contains bundle information.
+    * @return {object} promise.
+    */
+  loadBundles(config, bundle) {
+    let scripts = [];
+    let promises = [];
+    let styles = [];
+
+    if (bundle.scripts) {
+      scripts = filter(bundle.scripts, script => script.type === config.scriptType);
```

Comment:
> It is defined in config.scripts value. That value is passed to
if (scripts.length > 0 && config.scripts) {...
If the condition is true, all scripts will be loaded

`bundle.scripts` get filtered by type considering `config.scriptType`.  So, if user wants to load only one type of the scripts, he will pass it as either `meta` or `component`. 
My question is: what should be passed if user wants to load both `meta` and `component` type scripts?

Reply:
From what I have understood, the original issue says you can import either component or meta scripts, not both:
> Builder imports only meta scripts, while other two projects import only components scripts.

So I have not implemented that. Have I misunderstood it?

Reply:
Feature request stated following:
> Bundle injector utility that will be created here should be configurable (e.g. config object parameter or something like that):
> 
> - inject all or no style sheets
> - inject all or no scripts
> - inject only meta or only components scripts
> - disregard deploy script type mentioned in vue-cli-plugin-chameleon, that's legacy

So, options regarding scripts are: only meta, only component, all (both types) or no scripts.

Reply:
Ok, I have deleted the ```scripts``` property and changed ```scriptStyle``` to ```scriptStyles``` array.
It has 3 possibilities:
1. [] - empty array, import no scripts
2. ['meta'] or ['components'] - import only one type of script
3. ['meta', 'components'] - import both script types

Reply:
You mean `scriptTypes`? 😄

Reply:
Yes, lapsus calami 😄

Reply:
Ok, just two more things and we're good to go: make `bundle` the first parameter, that is the object of `loadBundle` method after all; rename `config.scriptTypes` to `config.scripts` (consistent with bundle manifest).

Reply:
Changed